includes/class-amp-story-post-type.php

```diff
@@ -341,7 +345,109 @@ public static function remove_classic_editor_link( $actions, $post ) {
 	}
 
 	/**
-	 * Filter the allowed tags for Kses to allow for amp-story children.
+	 * Filters allowed CSS attributes to include positioning, display, etc.
+	 *
+	 * @param array $attr Array of allowed CSS attributes.
+	 *
+	 * @return array Filtered array of allowed CSS attributes.
+	 */
+	public static function filter_safe_style_css( $attr ) {
+		global $wp;
+
+		$route = $wp->query_vars['rest_route'];
+
+		if ( ! $route ) {
+			return $attr;
+		}
+
+		$search = '\/' . self::POST_TYPE_SLUG . '\/';
+
+		// Only continue if it's a REST request for amp_story post type.
+		if ( ! preg_match( "/{$search}/i", $route ) ) {
+			return $attr;
+		}
+
+		$style_to_add = array(
+			'display',
+			'left',
+			'position',
+			'top',
+		);
+
+		foreach ( $style_to_add as $style ) {
+			$attr[] = $style;
+		}
+		return $attr;
+	}
+
+	/**
+	 * Filters the response before executing any REST API callbacks.
+	 *
+	 * Temporarily modifies post content during saving in a way that KSES
+	 * does not strip actually valid CSS from post content, making block content invalid.
+	 *
+	 * @todo Remove once core has better CSS parsing.
+	 *
+	 * @link https://core.trac.wordpress.org/ticket/37134
+	 *
+	 * @param WP_HTTP_Response|WP_Error $response Result to send to the client. Usually a WP_REST_Response or WP_Error.
+	 * @param array                     $handler  Route handler used for the request.
+	 * @param WP_REST_Request           $request  Request used to generate the response.
+	 *
+	 * @return WP_HTTP_Response|WP_Error The filtered response.
+	 */
+	public static function filter_rest_request_for_kses( $response, $handler, $request ) {
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$search = '\/' . self::POST_TYPE_SLUG . '\/';
+
+		$editable_request_methods = array_map( 'trim', explode( ',', WP_REST_Server::EDITABLE ) );
+
+		if ( ! in_array( $request->get_method(), $editable_request_methods, true ) || ! preg_match( "/{$search}/i", $request->get_route() ) ) {
+			return $response;
+		}
+
+		if ( ! current_user_can( 'edit_post', $request['id'] ) ) {
+			return $response;
+		}
+
+		$style_attr_values = [];
+
+		// Replace inline styles before KSES...
+		add_filter(
+			'pre_post_content',
+			static function ( $post_content ) use ( &$style_attr_values ) {
+				$post_content = preg_replace_callback(
+					'|style=\\\"([^"]*)\\\"|', // Because the post data is slashed.
```

Comment:
Just recalled that that this should be sure to include the `<` and `>` of the containing tag as otherwise this could match text content of elements, like if I write a story about `style` attributes.

So I suggest a pattern like:

```php
'#<\w+(?:-\w+)*\s[^>]*?style=\\\"([^"]*)\\\"[^>]+?>#'
```

In this way it will match the `style` attribute in the entire start tag.